### PR TITLE
[flutter_tools] remove mocks from cold test

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards #71511

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/cold_test.dart [hunk 1]
```diff
@@ -181,10 +163,42 @@ class MockFlutterDevice extends Mock implements FlutterDevice {
...
+  }
+
+  @override
+  Future<void> initLogReader() async { }
 }
-class MockDevice extends Mock implements Device {
-  MockDevice() {
-    when(isSupported()).thenReturn(true);
+
+class MockDevice extends Fake implements Device {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/cold_test.dart:178]: Can this be `FakeDevice`

File: packages/flutter_tools/test/general.shard/cold_test.dart [hunk 2]
```diff
@@ -171,7 +145,15 @@ void main() {
   });
 }
 
-class MockFlutterDevice extends Mock implements FlutterDevice {
+class MockFlutterDevice extends Fake implements FlutterDevice {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/cold_test.dart:148]: can this be `FakeFlutterDevice`

**Review by christopherfujino — APPROVED** [review]
two nits about `Mock` -> `Fake` classnames, but LGTM

File: packages/flutter_tools/test/general.shard/cold_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/cold_test.dart]: Done

File: packages/flutter_tools/test/general.shard/cold_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/cold_test.dart]: Done

**Review by christopherfujino — APPROVED** [review]
Ship it

**Pull request merged**